src/nanoemoji/color_glyph.py

```diff
@@ -326,8 +341,22 @@ def _painted_layers(
                 else:
                     raise ValueError(error_msg)
 
+        if success and len(paths) > 1:
+            # Don't create composite glyphs if components intersect each other and
+            # they have a transform which reverses their winding direction, or else this
+            # creates holes where they overlap as nonzero fill rule is applied by
+            # TrueType renderer: https://github.com/googlefonts/nanoemoji/issues/287
+            reverses_direction = [t.determinant() < 0 for t in transforms]
+            success = not any(
+                (
+                    (reverses_direction[i] or reverses_direction[j])
+                    and _intersect(paths[i], paths[j])
+                )
+                for i, j in combinations(range(len(paths)), 2)
+            )
```

Comment:
I think we should log when this kicks in

Reply:
done